homeassistant/components/risco/alarm_control_panel.py

```diff
@@ -82,8 +101,10 @@ def __init__(self, coordinator, partition_id, code, options):
         for state in self._ha_to_risco:
             self._attr_supported_features |= STATES_TO_SUPPORTED_FEATURES[state]
 
-    def _get_data_from_coordinator(self):
-        self._partition = self.coordinator.data.partitions[self._partition_id]
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id for this partition."""
+        raise NotImplementedError
```

Comment:
Nowadays we prefer using our short hand attributes, eg `_attr_unique_id`.

https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes

Reply:
Cool. What about the abstract base class? Do I simply remove this property?

Reply:
Yeah, I'd do that. It's only ourselves that will be using the base class so it's not that fragile.

Reply:
This causes a new problem though:
```
error: Argument 1 to <set> has incompatible type "Tuple[str, Optional[str]]"; expected "Tuple[str, str]"  [arg-type]
```

This is for the call to `self.unique_id` from `device_info()`

Reply:
Maybe pass the device unique_id as a parameter to this class and set `_attr_device_info` in init method?

Reply:
That will require every class to do that, and we might have new ones coming up when we split `bypassed` and `alarmed`.
Maybe I can initialize to an empty string in the base class, and use that instead of `self.unique_id`?

Reply:
Setting an empty string doesn't sound correct.

If the different children represent different devices with different unique ids it seems valid that they either provide the unique id to the parent which sets the device info, or that the children themselves set the device info.

Reply:
Yeah, I see your point about empty string not sounding right, but it's a bummer that we're adding additional coupling points just to make typing hints work in a situation where there are no real typing concerns.

Reply:
it's also a cycle because the unique id depends on `self.risco` which is set by one of the base class constructors. Argghh

Reply:
Let's see how it looks after the next push. It's easier to make suggestions then.

Reply:
Fair enough. Coming up. I had to add `or ''` just so I can commit